src/libs/kata-types/src/config/hypervisor/mod.rs

```diff
@@ -0,0 +1,1061 @@
+        }
+    }
+
+    /// Validate path of virtio-fs daemon, especially for annotations.
+    pub fn validate_virtiofs_daemon_path<P: AsRef<Path>>(&self, path: P) -> Result<()> {
+        validate_path_pattern(&self.valid_virtio_fs_daemon_paths, path)
+    }
+
+    fn adjust_virtio_fs(&mut self, _inline: bool) -> Result<()> {
+        resolve_path!(
+            self.virtio_fs_daemon,
+            "Virtio-fs daemon path {} is invalid: {}"
+        )?;
+        if self.virtio_fs_cache.is_empty() {
+            self.virtio_fs_cache = default::DEFAULT_VIRTIO_FS_CACHE_MODE.to_string();
+        }
+        if self.virtio_fs_is_dax && self.virtio_fs_cache_size == 0 {
+            self.virtio_fs_cache_size = default::DEFAULT_VIRTIO_FS_DAX_SIZE_MB;
+        }
+        if !self.virtio_fs_is_dax && self.virtio_fs_cache_size != 0 {
+            self.virtio_fs_is_dax = true;
+        }
+        Ok(())
+    }
+
+    fn validate_virtio_fs(&self, inline: bool) -> Result<()> {
+        if inline && !self.virtio_fs_daemon.is_empty() {
+            return Err(eother!(
+                "Executable path for inline-virtio-fs is not empty: {}",
+                &self.virtio_fs_daemon
+            ));
+        }
+        validate_path!(
+            self.virtio_fs_daemon,
+            "Virtio-fs daemon path {} is invalid: {}"
+        )?;
+
+        if self.virtio_fs_cache != "none"
+            && self.virtio_fs_cache != "auto"
+            && self.virtio_fs_cache != "always"
+        {
+            return Err(eother!(
+                "Invalid virtio-fs cache mode: {}",
+                &self.virtio_fs_cache
+            ));
+        }
+        if self.virtio_fs_is_dax && self.virtio_fs_cache_size == 0 {
+            return Err(eother!(
+                "Invalid virtio-fs DAX window size: {}",
+                &self.virtio_fs_cache_size
+            ));
+        }
+        Ok(())
+    }
+}
+
+/// Common configuration information for hypervisors.
+#[derive(Debug, Default, Deserialize, Serialize)]
+pub struct Hypervisor {
+    /// Path to the hypervisor executable.
+    #[serde(default)]
+    pub path: String,
+    /// List of valid annotations values for the hypervisor.
+    ///
+    /// Each member of the list is a path pattern as described by glob(3). The default if not set
+    /// is empty (all annotations rejected.)
+    #[serde(default)]
+    pub valid_hypervisor_paths: Vec<String>,
+
+    /// Hypervisor control executable path.
+    #[serde(default)]
+    pub ctlpath: String,
+    /// List of valid annotations values for the hypervisor control executable.
+    ///
+    /// Each member of the list is a path pattern as described by glob(3). The default if not set
+    /// is empty (all annotations rejected.)
+    #[serde(default)]
+    pub valid_ctlpaths: Vec<String>,
+
+    /// Control channel path.
+    #[serde(default)]
+    pub jailer_path: String,
+    /// List of valid annotations values for the hypervisor.
```

Comment:
Comments are the same as `valid_hypervisor_paths`, is it right or just a copy miss?

Reply:
fixed